### PR TITLE
✨ feat : password based authentication system

### PR DESCRIPTION
**Pull request by arkojeet-bera**
> Auth UI created (responsive).
> Routes nested under /auth/{...}
> Sign-Up Form Functionality working with proper validation according to Backend Schema.

@Bismay5467 Please have a look!

### CODE REVIEW

File: client/src/types/authTypes.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { z } from 'zod';
+import { newUserInputSchema } from '@/constants/authSchema';
+
+export type SignUpFormFields = z.infer<typeof newUserInputSchema>;
```
Comment (Bismay5467) [inline client/src/types/authTypes.ts]: Minor suggestion : prefix types with T and interfaces with I to enhance code readability.

File: client/src/pages/Auth/Signup/Signup.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    <AccountForm register={register} />,
+    <OTPForm register={register} />,
+  ]);
+
+  const onSubmit: SubmitHandler<SignUpFormFields> = async (formData) => {
+    const payload: SignUpFormFields = {
+      ...formData,
+      role: 'USER',
+      actionType: 'GENERATE',
+    };
```
Comment (Bismay5467) [inline client/src/pages/Auth/Signup/Signup.tsx]: I am assuming this is a temporary payload created for a sake of testing? Bcoz we have a actionType as 'VERIFY' too.

File: client/src/pages/Auth/Signup/MultiStepForm/AccountForm.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { UseFormRegister } from 'react-hook-form';
+import { FaEye, FaEyeSlash } from 'react-icons/fa';
+import { MdOutlineEmail } from 'react-icons/md';
+
+export default function AccountForm({
+  register,
+}: {
+  register: UseFormRegister<SignUpFormFields>;
+}) {
+  const [eyeOff, setEyeOff] = useState(true);
```
Comment (Bismay5467) [inline client/src/pages/Auth/Signup/MultiStepForm/AccountForm.tsx]: Minor suggestion : define hooks as `useState<boolean>(true)`. This provides better type safety.

**Comment by Bismay5467** [comment]
@simply-arko suggestions: 
- Instead of just focusing on the fields in case of wrong input we can have a label for error message indicating what is wrong in the input. 
- I noticed you've installed the input-otp npm library. I was hoping to see the UI provided by that library, as it looks much cooler IMO. 😅
- you may check out `clamp()` in CSS - for making the text responsive. 

Other than that great work!!

The migration from TRPC to REST APIs is nearly complete. I'll be merging a PR by EOD. Once that's done, you can integrate the APIs into the frontend.

File: client/src/pages/Auth/Signup/Signup.tsx (hunk 2 above)
Comment (ghost) [reply client/src/pages/Auth/Signup/Signup.tsx]: Yeah it's temporary

**Comment by ghost** [comment]
@Bismay5467 the input OTP library is having some problems integrating with react-hook-form (specifically `register` is not working here). I'll be working on that and labelling the error messages in-place soon once I'm done with integrating the APIs.

**Comment by ghost** [comment]
@Bismay5467 Please note that the previous PRs as not ready for review. There is some level of code re-factoring still required. Let's just finish up search and merge it first and then I'll make the final PR for review. That way, I can minimise any merge conflicts (if any)!

**Comment by ghost** [comment]
@Bismay5467 I've implemented the login functionality. The user can login and it's being made persistent with context API. The navbar is updated conditionally to reflect user avatar upon login! Please have a look!

**Comment by ghost** [comment]
@Bismay5467 As discussed earlier, you can remove the response cookie from reset controller (case: `RESET`) as we are anyway re-routing the user to login page. Requesting suggestions and review!

**Pull request merged**